packages/flutter_tools/lib/src/android/gradle.dart

```diff
@@ -340,13 +355,40 @@ Future<void> buildGradleProject({
       // Fall through. Pretend it's v1, and just go for it.
     case FlutterPluginVersion.v1:
       return _buildGradleProjectV1(project, gradle);
-    case FlutterPluginVersion.managed:
-      // Fall through. Managed plugin builds the same way as plugin v2.
+    case FlutterPluginVersion.v3:
+      return _buildGradleProjectV3(project, gradle, buildInfo, target, isBuildingBundle);
     case FlutterPluginVersion.v2:
       return _buildGradleProjectV2(project, gradle, buildInfo, target, isBuildingBundle);
   }
 }
 
+// TODO(mklim): Fix the obvious bugs.
+// - Do we need logic for passing along build architecture, local engine, etc?
+// - Make this smart enough to not rebuild the plugin every time?
+Future<void> buildPluginAAR(Plugin plugin, String gradle, String assembleTask) async {
```

Comment:
For this and any other procedures that we add, can we include ample comments about the purpose, preconditions, and postconditions for each one?

Reply:
This is going away.